src/main/java/edu/harvard/iq/dataverse/UserNameValidator.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package edu.harvard.iq.dataverse;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+/**
+ *
+ * @author sarahferry
+ * Modeled after PasswordValidator and EMailValidator
+ */
+
+public class UserNameValidator implements ConstraintValidator<ValidateUserName, String> {
+    @Override
+    public void initialize(ValidateUserName constraintAnnotation) {
+
+    }
+    
+    @Override
+    public boolean isValid(String value, ConstraintValidatorContext context) {
+        return isUserNameValid(value, context);
+    }
+
+    /**
+     * Here we will validate the username
+     *
+     * @param username
+     * @return boolean 
+     */
+    public static boolean isUserNameValid(final String username, ConstraintValidatorContext context) {
```

Comment:
The matcher throws a null pointer exception if it encounters null, so I added it.